libs/ml/include/ml/core/graph.hpp

```diff
@@ -523,6 +526,37 @@ bool Graph<TensorType>::SetRegularisation(std::string node_name, RegPtrType regu
   return true;
 }
 
+/**
+ * Set variable freezing for all trainables in graph
+ * @tparam TensorType
+ * @param frozen_state true=freeze variables, false=unfreeze variables
+ */
+template <typename TensorType>
+void Graph<TensorType>::SetFrozenState(bool frozen_state)
+{
+  for (auto &t : trainable_lookup_)
```

Comment:
this doesn't work for any interesting graphs. That's because trainable_lookup_ doesn't store all of the trainables; only the trainables in THIS graph.
You need to call GetTrainables which makes a recursive call to collect all the trainables.